### PR TITLE
Add solutions to homework one

### PR DESCRIPTION
**Pull request by rajgthub**
I attach my solutions to homework-exercise one here with this pull request. Please have a look at the answers and give your feedback when you get time. Thanks  very much for  spending your time on checking my work! I have not focused much on the styling bit.... I should have separated forms into component! I also used state to add a little bit of interaction to the UI. If you want me to do without states, I can do it again. Let me know your thoughts!

### CODE REVIEW

File: src/components/table.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      <td>{surname}</td>
+      <td>{email}</td>
+      <td>{roomId}</td>
+      <td>{checkInDate}</td>
+      <td>{checkOutDate}</td>
+      <td>{dateDifference(checkInDate, checkOutDate)}</td>
+    </tr>
+  </tbody>
+);
+export { TableHeader, TableRow };
```
Comment (40thieves) [inline src/components/table.js]: Nice! In this case I agree with your decision to keep the components in the same file as they are related to each other. But the convention generally is to put components in separate files.

File: src/containers/Bookings.js [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
-
+import React, { Component } from "react";
+import Search from "../components/Search.js";
+import Results from "../components/Results.js";
+import rsults from "../data/fakeBookings.json";
 export default class Bookings extends Component {
+  state = { rsults: undefined };
   search = () => {
-    console.info('TO DO!');
+    this.setState({ rsults});
```
Comment (40thieves) [inline src/containers/Bookings.js]: 👍 We're going to flesh out this function more in upcoming exercises once we've looked at state in class.

File: src/components/table.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    <tr>
+      <td>{id}</td>
+      <td>{title}</td>
+      <td>{firstName}</td>
+      <td>{surname}</td>
+      <td>{email}</td>
+      <td>{roomId}</td>
+      <td>{checkInDate}</td>
+      <td>{checkOutDate}</td>
+      <td>{dateDifference(checkInDate, checkOutDate)}</td>
```
Comment (40thieves) [inline src/components/table.js]: Nice use of a function to clean up here!

File: src/components/Results.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import "../Results.css";
+import { TableHeader, TableRow } from "./table";
+const Results = ( {results} ) => {
+    console.log('testing...', results)
+  if (results && results.length > 0) {
+    return (
+      <table className="Results">
+        <TableHeader />
+        {results.map((result, key) => <TableRow key= {key} {...result} />)}
```
Comment (40thieves) [inline src/components/Results.js]: Good that you included a `key` 👏

File: src/components/Search.js [hunk 5]
```diff
@@ -10,8 +10,28 @@ const Search = props => (
...
+              id="customerId"
+              className="form-control"
+              placeholder="Customer Id"
+            />
+            <Buttton label="Search IDs" search= {search}/>
+          </div>
+        </div>
+      </div>
+      {/* search by name*/}
+      <div className="col">
```
Comment (40thieves) [inline src/components/Search.js]: This isn't part of the exercises, but do you notice there's some duplication here - what could you do to improve that?

**Comment by rajgthub** [comment]
Thanks very much for the feedback and the time you spent on checking my solutions even if you were so busy with your work! I will reply to your comments when I get home.

File: src/components/table.js (hunk 1 above)
Comment (rajgthub) [reply src/components/table.js]: Ok

File: src/components/Search.js (hunk 5 above)
Comment (rajgthub) [reply src/components/Search.js]: You are right! I have done the changes as you pointed out. I hope I did not make it more complicated!

**Comment by rajgthub** [comment]
Hi Ali,

Thanks very much for checking my solutions. Yes, I will delete the table.js
as I am not using it anymore.
Best wishes,
Raju

File: src/containers/Bookings.js [hunk 6]
```diff
@@ -1,13 +1,36 @@
...
+          return res;
+        } else {
+          throw new Error("HTTP error");
+        }
+      })
+      .then(res => res.json())
+      .then(({ results }) => {
+        console.log("kehtnn", results);
+        bookings = results;
+        this.setState({ isLoading: false, filteredRsults: results });
```
Comment (40thieves) [inline src/containers/Bookings.js]: One quick tip - it's probably a good idea to clear out any errors you have in state when you receive a successful response. That way you don't accidentally show an error message even if there is a successful response

File: src/containers/Bookings.js [hunk 7]
```diff
@@ -57,27 +80,35 @@ export default class Bookings extends Component {
...
+        ) : (
+          !this.state.error && "No results found!!"
+        )}
+      </div>
+    );
+    const templateLoading = <h1>successfully completed!</h1>
+    /* I am using two different identifier: 
+    database has names like first_name and API returns as firstName; 
+    not enogh time! 
+    I will change it later*/
```
Comment (40thieves) [inline src/containers/Bookings.js]: 👍 It looks like you're on the right track though!

**Pull request closed without merging**